src/main/java/com/mackervoy/calum/mud/behaviour/task/AbstractTaskActor.java

```diff
@@ -95,18 +122,74 @@ protected Resource getFirstResourceMatchingType(Model request, Resource type) {
 		}
 	}
 	
-	protected void commitToDB() {
-		this.model.add(this.task.getModel());
+	/**
+	 * @return the LocalDateTime when this task will end, or null if it doesn't have one set 
+	 * Default uses the Time ontology, so it's useful to override this if you want to use something else
+	 */
+	public Optional<LocalDateTime> getTaskEndTime() {
+		Resource end = this.task.getPropertyResourceValue(Time.hasEnd);
+		if(end == null) return Optional.empty();
 		
-		Dataset dataset = TDB2Factory.connectDataset(this.taskDatasetItem.getFileLocation());
-		try {
-			dataset.begin(ReadWrite.WRITE);
-			Model out = dataset.getDefaultModel();
-			out.add(this.model);
-			out.commit();
+		return Optional.of(Time.instantToLocalDateTime(end));
```

Comment:
Using local time will tie us to the local timezone of the server, and it doesn't look like it'll expose that timezone information in the outputted string (meaning we couldn't tell the user the time in their timezone, or what timezone it's in without also exposing the timezone of the server). 

Can we use UTC on the server, and then convert it out into a users timezone on the front end instead?

Reply:
I think using UTC as a standard makes sense :+1: 

Also made me think of this one https://github.com/Multi-User-Domain/vocab/issues/8